lib/live_view_native/swiftui/styles/utility.ex

```diff
@@ -0,0 +1,174 @@
+defmodule LiveViewNative.SwiftUI.UtilityStyles do
+  use LiveViewNative.Stylesheet, :swiftui
+  @output false
+
+  @moduledoc """
+  Tailwind-style utility classes for LiveView Native.
+
+  ## Modifier Translation
+  Modifiers can easily be translated to classes.
+
+  1. Write the name of the modifier in `snake-case`
+  2. Separate any arguments with a dash (`-`)
+  3. Use labels for any named arguments (`key:value`)
+
+  | Modifier | Class |
+  | -------- | ----- |
+  | `padding(.leading, 32)` | `padding-leading-32` |
+  | `frame(width: 100, height: 100)` | `frame-width:100-height:100` |
+  | `buttonStyle(.borderedProminent)` | `button-style-borderedProminent` |
+
+  ### Attribute References
+  To reference an attribute value, use `[attr(attribute-name)]`:
+
+  ```html
+  <Group class="offset-y:[attr(offset)]" offset="5">
+  ```
+
+  ### Complex Arguments
+  Complex types (such as a gradient `ShapeStyle`) can be provided by writing out the full syntax as an argument.
+
+  ```html
+  <Rectangle class="fg-LinearGradient(colors:[.red,.blue],startPoint:.leading,endPoint:.trailing)" />
+  ```
+
+  > [!NOTE]
+  > Spaces cannot be used within class names.
+
+  ### Spaces and Underscores
+  To include a space in a string argument, use and underscore `['Hello,_world!']`.
+  Any underscores will be replaced with a space.
+
+  ```html
+  <Group class="navigation-title-['Hello_World']">
+  ```
+
+  If you need to include an underscore character, escape it with `\_`.
+
+  ```html
+  <Group class="navigation-title-['Hello\_World']">
+  ```
+
+  ### Shorthand
+  Some aliases are provided to make common classes easier to use.
+
+  | Alias | Expansion |
+  | ----- | --------- |
+  | `px` | `padding-horizontal` |
+  | `py` | `padding-vertical` |
+  | `pt` | `padding-top` |
+  | `pr` | `padding-trailing` |
+  | `pb` | `padding-bottom` |
+  | `pl` | `padding-leading` |
+  | `pl` | `padding-leading` |
+  | `p` | `padding` |
+  | `w-` | `frame-width:` |
+  | `h-` | `frame-height:` |
+  | `min-w-` | `frame-minWidth:` |
+  | `max-w-` | `frame-maxWidth:` |
+  | `min-h-` | `frame-minHeight:` |
+  | `max-h-` | `frame-maxHeight:` |
+  | `fg` | `foreground-style` |
+  | `bg` | `background` |
+  | `overlay--` | `overlay-content::` |
+  | `bg--` | `background-content::` |
+  | `mask--` | `mask-mask::` |
+  | `toolbar--` | `toolbar-content::` |
+  | `safe-area-inset--` | `safe-area-inset-content::` |
+  """
+
+  @modifier_names (File.read!("lib/live_view_native/swiftui/styles/modifier.names") |> String.split("\n", trim: true))
+    ++ ["stroke", "mask"]
```

Comment:
@carson-katri I was curious why these two were appended? I'm guessing the first list was from code gen?

Reply:
Here's a complete list you can use, and remove the appended ones:

```
accessibilityAction
accessibilityActions
accessibilityChildren
accessibilityIgnoresInvertColors
accessibilityRepresentation
accessibilityShowsLargeContentViewer
alert
allowsHitTesting
allowsTightening
animation
aspectRatio
autocorrectionDisabled
background
backgroundStyle
badge
baselineOffset
blendMode
blur
bold
border
brightness
buttonBorderShape
buttonStyle
clipShape
clipped
colorInvert
colorMultiply
compositingGroup
confirmationDialog
containerRelativeFrame
containerShape
contentShape
contentTransition
contextMenu
contrast
controlGroupStyle
controlSize
coordinateSpace
datePickerStyle
defaultScrollAnchor
defaultWheelPickerItemHeight
defersSystemGestures
deleteDisabled
dialogSuppressionToggle
digitalCrownAccessory
disabled
drawingGroup
dynamicTypeSize
fileDialogCustomizationID
fileDialogImportsUnresolvedAliases
fill
findDisabled
findNavigator
fixedSize
flipsForRightToLeftLayoutDirection
focusEffectDisabled
focusScope
focusSection
focusable
font
fontDesign
fontWeight
fontWidth
foregroundStyle
formStyle
frame
fullScreenCover
gaugeStyle
geometryGroup
gesture
grayscale
gridCellAnchor
gridCellColumns
gridCellUnsizedAxes
gridColumnAlignment
groupBoxStyle
headerProminence
help
hidden
highPriorityGesture
horizontalRadioGroupLayout
hoverEffect
hoverEffectDisabled
hueRotation
ignoresSafeArea
imageScale
indexViewStyle
inspector
inspectorColumnWidth
interactionActivityTrackingTag
interactiveDismissDisabled
intersection
invalidatableContent
italic
kerning
keyboardShortcut
keyboardType
labelStyle
labeledContentStyle
labelsHidden
layoutPriority
lineIntersection
lineLimit
lineSpacing
lineSubtraction
listItemTint
listRowBackground
listRowHoverEffect
listRowHoverEffectDisabled
listRowInsets
listRowSeparator
listRowSeparatorTint
listRowSpacing
listSectionSeparator
listSectionSeparatorTint
listSectionSpacing
listStyle
luminanceToAlpha
mask
matchedGeometryEffect
menuIndicator
menuOrder
menuStyle
minimumScaleFactor
monospaced
monospacedDigit
moveDisabled
multilineTextAlignment
navigationBarBackButtonHidden
navigationBarTitleDisplayMode
navigationDestination
navigationSplitViewColumnWidth
navigationSplitViewStyle
navigationSubtitle
navigationTitle
offset
onAppear
onDeleteCommand
onDisappear
onExitCommand
onHover
onLongPressGesture
onLongTouchGesture
onMoveCommand
onPlayPauseCommand
onSubmit
onTapGesture
opacity
overlay
padding
persistentSystemOverlays
pickerStyle
popover
position
preferredColorScheme
prefersDefaultFocus
presentationBackground
presentationBackgroundInteraction
presentationCompactAdaptation
presentationContentInteraction
presentationCornerRadius
presentationDetents
presentationDragIndicator
previewDisplayName
privacySensitive
progressViewStyle
projectionEffect
redacted
refreshable
renameAction
renderingMode
replaceDisabled
resizable
rotation
rotation3DEffect
rotationEffect
safeAreaInset
saturation
scale
scaleEffect
scaledToFill
scaledToFit
scenePadding
scrollBounceBehavior
scrollClipDisabled
scrollContentBackground
scrollDisabled
scrollDismissesKeyboard
scrollIndicators
scrollIndicatorsFlash
scrollPosition
scrollTargetBehavior
scrollTargetLayout
searchCompletion
searchDictationBehavior
searchPresentationToolbarBehavior
searchScopes
searchSuggestions
searchable
selectionDisabled
shadow
sheet
simultaneousGesture
speechAdjustedPitch
speechAlwaysIncludesPunctuation
speechAnnouncementsQueued
speechSpellsOutCharacters
statusBarHidden
strikethrough
stroke
submitLabel
submitScope
subtracting
swipeActions
symbolEffect
symbolEffectsRemoved
symbolRenderingMode
symbolVariant
symmetricDifference
tabItem
tabViewStyle
tableStyle
textCase
textContentType
textEditorStyle
textFieldStyle
textInputAutocapitalization
textScale
textSelection
tint
toggleStyle
toolbar
toolbarBackground
toolbarColorScheme
toolbarRole
toolbarTitleDisplayMode
toolbarTitleMenu
touchBarCustomizationLabel
touchBarItemPrincipal
tracking
transform
transformEffect
transition
truncationMode
underline
union
unredacted
zIndex
```

Reply:
I was thinking that the `modifier.names` file could stay as just a result of codegen. If you have a script to generate that we could add to this repo so we can run it from time to time and update the modifier list

Reply:
I could add something to the `ModifierGenerator`, which is what I used to generate the list (plus some `jq` to manipulate it).

Reply:
I've added a `list` subcommand to the modifier generator in #1250, which will allow the file to be generated.